corosync_metrics.go

```diff
@@ -21,33 +20,27 @@ var (
 		"ring_errors_total": NewMetricDesc("corosync", "ring_errors_total", "Total number of corosync ring errors", nil),
 		"quorum_votes":      NewMetricDesc("corosync", "quorum_votes", "Cluster quorum votes; one line per type", []string{"type"}),
 	}
-
-	corosyncTools = map[string]string{
-		"quorumtool": "/usr/sbin/corosync-quorumtool",
-		"cfgtool":    "/usr/sbin/corosync-cfgtool",
-	}
 )
 
-func NewCorosyncCollector() (*corosyncCollector, error) {
-	for _, toolPath := range corosyncTools {
-		fileInfo, err := os.Stat(toolPath)
-		if os.IsNotExist(err) {
-			return nil, errors.Wrapf(err, "'%s' not found", toolPath)
-		}
-		if (fileInfo.Mode() & 0111) == 0 {
-			return nil, errors.Errorf("'%s' is not executable", toolPath)
-		}
+func NewCorosyncCollector(cfgToolPath string, quorumToolPath string) (*corosyncCollector, error) {
+	err := CheckExecutables(cfgToolPath, quorumToolPath)
+	if err != nil {
+		return nil, errors.Wrap(err, "external executable check failed")
```

Comment:
are we errorring here much generic? 

If we are wrapping error we should add  a more precise context on corosync

Reply:
The message of the wrapped error will also show which executable is failing, so it will not be that generic! This is an example of the final result when launching the exporter:

```
$ ./ha_cluster_exporter 
INFO[0000] Using config file: /home/etienne/work/SUSE/ha_cluster_exporter/ha_cluster_exporter.yaml 
WARN[0000] Could not initialize Pacemaker collector: external executable check failed: stat /usr/sbin/crm_mon: no such file or directory 
WARN[0000] Could not initialize Corosync collector: external executable check failed: stat /usr/sbin/corosync-cfgtool: no such file or directory 
WARN[0000] Could not initialize SBD collector: external executable check failed: stat /usr/sbin/sbd: no such file or directory 
WARN[0000] Could not initialize DRBD collector: external executable check failed: stat /usr/sbin/drbdsetup: no such file or directory 
INFO[0000] Serving metrics on 0.0.0.0:9002 
```

I think it's verbose enough, no?

Reply:
yeah i resolved. ignore this.